flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -126,10 +126,13 @@ public FlinkCatalog(
   @Override
   public void open() throws CatalogException {
     // Create the default database if it does not exist.
-    try {
-      createDatabase(getDefaultDatabase(), ImmutableMap.of(), true);
-    } catch (DatabaseAlreadyExistException e) {
-      // Ignore the exception if it's already exist.
+    String defaultDatabase = getDefaultDatabase();
+    if (!databaseExists(defaultDatabase)) {
+      try {
+        createDatabase(getDefaultDatabase(), ImmutableMap.of(), true);
+      } catch (DatabaseAlreadyExistException e) {
```

Comment:
this is also a little weird since we already did the `dataExists` check earlier. I understand why this is needed to handle race condition when multiple jobs can run this code path concurrently.

The first attempt of ignoring `ForbiddenException` was still not ideal to me because it can mask real issue that default database wasn't created due to permission issue.

I checked the 'HiveCatalog` and `GenericInMemoryCatalog` implementation from Flink code. They don't try to create the default database in the `open` method. Flink `HiveCatalog` just assert that default database exists. That behavior seems reasonable to me.

cc @pvary

Reply:
Are you suggesting removing the create database call? The person that initially reported this issue, was very supprised that this call was in there. So I think that removing the call seems reasonable to me.

Reply:
Yea, I was thinking about remove the create database call and just add a `Preconditions` check that the default database exists. that is also consistent with Flink's `HiveCatalog` implementation.

The `Preconditions` check is also debatable. I am not very clear about the expectation of catalog. is the `default` database always expected/required? not sure if @rdblue has any input.

Reply:
I'd be okay with removing the check, although I think as long as attempting to create it doesn't cause a failure it is probably slightly better to create it.

Reply:
@Fokko I am still leaning toward removing the create. This is a Flink `Catalog` impl (not Iceberg catalog). I think we can stick with Flink `HiveCatalog` style, where `open` method doesn't create the default database automatically.

Reply:
@stevenzwu I agree. I would not expect writing data would also create a database. I'll update the PR

Reply:
I've updated the tests as well. I think this one is good to go @stevenzwu __